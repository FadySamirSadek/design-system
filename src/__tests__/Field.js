import React from 'react'
import renderer from 'react-test-renderer'
import { Field } from '..'

describe('Field', () => {
  test('renders', () => {
    const json = renderer
      .create(<Field id="renders" label="Renders" />)
      .toJSON()
    expect(json).toMatchSnapshot()
  })
})
