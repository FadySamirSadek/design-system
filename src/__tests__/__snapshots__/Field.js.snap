// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field renders 1`] = `
.c0 {
  display: block;
  margin-bottom: 8px;
  font-size: 16px;
  color: #384046;
}

.c0 a {
  color: #2d9ee4;
  text-decoration: underline;
}

.c1 {
  margin-bottom: 2px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  vertical-align: middle;
  max-width: 32rem;
  min-height: 36px;
  line-height: inherit;
  font-family: inherit;
  background-color: transparent;
  border-radius: 4px;
  border-width: 1px;
  border-style: solid;
  border-color: #dde1e4;
  -webkit-transition: 0.125s box-shadow ease-out;
  transition: 0.125s box-shadow ease-out;
  font-size: 16px;
  margin: 0px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  width: 100%;
  color: inherit;
  background-color: transparent;
}

.c2::-ms-clear {
  display: none;
}

.c2::-ms-clear {
  display: none;
}

.c2::-ms-clear {
  display: none;
}

.c2::-ms-clear {
  display: none;
}

.c2:focus {
  outline: none;
  border-color: #2d9ee4;
  box-shadow: 0 0 0 2px #add9f5;
}

.c2[type='select'] {
  background: #fff url("data:image/svg+xml;charset=utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 4 5'%3E%3Cpath fill='%23606e77' d='M2 0L0 2h4zm0 5L0 3h4z'/%3E%3C/svg%3E") no-repeat right .75rem center;
  background-size: .5rem;
}

<label
  className="text c0"
  color="black"
  fontSize={2}
  id="renders"
>
  <div
    className="c1"
    wrap={true}
  >
    Renders
  </div>
  <input
    className="c2"
    color="inherit"
    fontSize={2}
    name="renders"
    placeholder={undefined}
    type="text"
  />
</label>
`;

exports[`Field with error 1`] = `
.c0 {
  display: block;
  margin-bottom: 8px;
  font-size: 16px;
  color: #384046;
}

.c0 a {
  color: #2d9ee4;
  text-decoration: underline;
}

.c1 {
  margin-bottom: 2px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  vertical-align: middle;
  max-width: 32rem;
  min-height: 36px;
  line-height: inherit;
  font-family: inherit;
  background-color: transparent;
  border-radius: 4px;
  border-width: 1px;
  border-style: solid;
  border-color: #dde1e4;
  -webkit-transition: 0.125s box-shadow ease-out;
  transition: 0.125s box-shadow ease-out;
  font-size: 16px;
  margin: 0px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  width: 100%;
  color: inherit;
  background-color: transparent;
}

.c3::-ms-clear {
  display: none;
}

.c3::-ms-clear {
  display: none;
}

.c3::-ms-clear {
  display: none;
}

.c3::-ms-clear {
  display: none;
}

.c3:focus {
  outline: none;
  border-color: #2d9ee4;
  box-shadow: 0 0 0 2px #add9f5;
}

.c3[type='select'] {
  background: #fff url("data:image/svg+xml;charset=utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 4 5'%3E%3Cpath fill='%23606e77' d='M2 0L0 2h4zm0 5L0 3h4z'/%3E%3C/svg%3E") no-repeat right .75rem center;
  background-size: .5rem;
}

.c2 {
  font-size: 14px;
  margin: 0px;
  margin-left: 4px;
  margin-top: 0px;
  margin-bottom: 0px;
  color: #b52434;
  font-weight: normal;
  text-transform: lowercase;
}

.c2:before {
  content: 'â€” ';
}

<label
  className="text c0"
  color="black"
  fontSize={2}
  id="error"
>
  <div
    className="c1"
    wrap={true}
  >
    Error
    <p
      className="error c2"
      color="error"
      f={1}
      ml={1}
      my={0}
    >
      error message
    </p>
  </div>
  <input
    className="c3"
    color="inherit"
    fontSize={2}
    name="error"
    placeholder={undefined}
    type="text"
  />
</label>
`;
